packages/profile-sync-controller/src/controllers/user-storage/encryption/encryption.ts

```diff
@@ -174,12 +206,26 @@ class EncryptorDecryptor {
     }
 
     const newSalt = salt ?? randomBytes(SCRYPT_SALT_SIZE);
-    const newKey = scrypt(password, newSalt, {
-      N: o.N,
-      r: o.r,
-      p: o.p,
-      dkLen: o.dkLen,
-    });
+
+    let newKey: Uint8Array;
+
+    if (nativeScryptCrypto) {
+      newKey = await nativeScryptCrypto(
```

Comment:
If I understand it right, the API for both of the scrypt flavors are identical as relevant for our usage in this module?

If so, I could identify a room for simplification:
Instead of having the constructor take an optional `nativeScrypt` parameter and then passing it on via parameters, could use a Dependency Injection pattern where any compatible `scrypt` implementation can be passed in. The default when not set could still be set to `@noble/hashes/scrypt` or `@noble/hashes/scrypt.scryptAsync`. The value could be set as `this.#scrypt`. Then we could call that directly here without this function needing to be aware of which implementation is actually used. WDYT?

Reply:
Yes I agree, there is significant prop drilling done

```
UserStorageController
> Encryption.encryptString()
> Encryption.encryptStringV1()
> Encryption.getOrGenerateScryptKey() // finally we check and use the native scrypt :)
```

The above Encryption class was treated and exported as a singleton, but I think we can make an exception for the controller to make it accept a custom instance.
OR allow the singleton instance to store and save a `Scrypt` implementation which can be overwritten (not just in the constructor).

I'll try to do some small cleanup in this PR (keep mobile unblocked), but will ticket a larger refactor.
- This package supports subpath exports, I'm tempted to keep both web and mobile scrypt implementations here as a SOT, and platforms can choose which to use -- e.g. `import {} from '@profile-sync-controller/mobile'`

Reply:
Hmm I've ticked this, will do this in a follow-up PR. There are some intricacies to this (changing this from an exported singleton to an instance tied to the Controller, or add a nice way of updating the crypto context)